app/controllers/appliances_controller.rb

```diff
@@ -68,8 +69,14 @@ def set_appliance
       @appliance = Appliance.find(params[:id])
     end
 
+    def set_kitchens
+      @kitchens = Kitchen.all
+    end
+
     # Only allow a list of trusted parameters through.
     def appliance_params
-      params.require(:appliance).permit(:name, :category, :status, :description, :kitchen_id)
+      parameters = params.require(:appliance).permit(:name, :category, :status, :description, :kitchen)
+      parameters[:kitchen] = Kitchen.find(parameters[:kitchen])
```

Comment:
Is this neceseary? We send the id of the kitchen from the form, and if we keep the naming convention, and call `kitchen` to `kitchen_id` Rails will automaticaly find the correspondig kitchen.

Reply:
Tried it, says it needs valid kitchen. :(

Reply:
I tried it locally and it should work. Can you push up the code that throw the error?

Reply:
Seems like (after long research) `kitchen_id` apparently does not find the corresponding kitchen implicitly, so the next commit solves the explicitly the issue.